src/thing/WiFiManager.cpp

```diff
@@ -0,0 +1,69 @@
+#include "thing/WiFiManager.h"
+
+#include <ESP8266WiFi.h>
+#include <DNSServer.h>
+
+
+namespace thing {
+namespace {
+
+const short kDnsPort = 53;
+
+}  // namespace
+
+WiFiManager::WiFiManager() : debug_([](const char*){}) {}
+
+bool WiFiManager::StartAP() {
+  // Use arduino string, casting an int to std::string with arduino C++ support
+  // isn't easy.
+  String ssid("FireThing-");
+  ssid += ESP.getChipId();
+
+  WiFi.mode(WIFI_AP_STA);
+  if (!WiFi.softAP(ssid.c_str())) {
+    return false;
+  }
+
+  debug_((String("WiFi AP : ") + ssid).c_str());
+  debug_((String("Wifi AP IP : ") + WiFi.softAPIP().toString()).c_str());
```

Comment:
that does sound pretty compelling. I spawned https://github.com/firebase/firebase-arduino/issues/233 to explore this, don't see any point blocking this PR on it though as this will be the basis for that work.